uberfire-workbench/uberfire-workbench-client/src/main/java/org/uberfire/client/mvp/ActivityBeansCache.java

```diff
@@ -63,15 +62,31 @@
      * All active activities that are splash screens.
      */
     private final List<SplashScreenActivity> splashActivities = new ArrayList<SplashScreenActivity>();
-    @Inject
     private SyncBeanManager iocManager;
-    @Inject
     private Event<NewPerspectiveEvent> newPerspectiveEventEvent;
-    @Inject
     private Event<NewWorkbenchScreenEvent> newWorkbenchScreenEventEvent;
+    private CategoriesManagerCache categoriesManagerCache;
+    private ResourceTypeManagerCache resourceTypeManagerCache;
```

Comment:
@adrielparedes why keep two instances of ActivityAndMetaInfo? I would delete 
private final List<ActivityAndMetaInfo> resourceActivities = new ArrayList<ActivityAndMetaInfo>();

and keep all the calls inside the resourceTypeManagerCache